### PR TITLE
OOP school library: refactor your code

### PR DESCRIPTION
**Pull request by ichala**
- Make sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- Follow SOLID principles.
- Pay special attention to the "Single-responsiblity Principle".
- Make sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).
- Extract some pieces of logic into separate class(es).

### CODE REVIEW

File: create.rb [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        puts
+        puts 'No record were found for the given ID'
+      end
+    end
+  end
+
+  def create_person
+    print 'press 1 to create a student, press 2 to create a teacher : '
+    option = gets.chomp
+
```
Comment (zhadier) [inline create.rb]: - [x] Great job on the overall refactor. One thing you do need to change to follow the Single responsibility principle is to ensure that all logic relating to requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input. You can refer to [this lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) to remind yourself about SOLID principles. The purpose of create_person should only be to create the person after it gets relevant data from another function/ class.

File: create.rb [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    puts 'Database is empty! Add a person.' if @persons.empty?
+    @persons.each do |person|
+      puts "[#{person.class.name}] Name: #{person.name}, Age: #{person.age}, id: #{person.id}"
+    end
+  end
+
+  def list_all_rentals
+    puts 'To see rentals enter the person ID: '
+    id = gets.chomp.to_i
+
```
Comment (zhadier) [inline create.rb]: - [x] Please ensure all data relating to getting and validating data from a person have a class/function of their own. So list all rentals will only list the rentals after it gets data from another function. You might have a function that takes a message as an input and displays it to the response. It then takes the answer, validates it and returns it to the function that called it.

File: create.rb [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  end
+
+  def create_student
+    puts 'create a new student'
+    print 'Enter student age: '
+    age = gets.chomp.to_i
+    print 'Enter name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
```
Comment (zhadier) [inline create.rb]: - [x] Kindly do the same thing with the input handling here and all preceding functions aswell.

**Review by zhadier — CHANGES_REQUESTED** [review]
# Hi @ichala & @Ogaga01  👋 

Good job so far! 🌤️ 
There are some issues that you still need to work on to go to the next project but you are almost there! 🍓 



## Highlights
✔️ No linter errors
✔️ GitFlow was set up correctly
✔️ Professional pull request title and description


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question @zhadier so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rotimiazeez — APPROVED** [review]
# Hi Team 👍🏽 , 

### Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**